### PR TITLE
move strategic merge patch transformer to a builtin transformer

### PR DESCRIPTION
**Pull request by Liujingfang1**
keep the previous unit tests for strategic merge patch transformer and also add some new ones

### CODE REVIEW

**Comment by damienr74** [comment]
Nevermind, In trying this locally I realize that git will mark the file as deleted if you modify it in the same commit.

File: k8sdeps/transformer/patch/transformer_test.go [hunk 1]
```diff
@@ -1,579 +0,0 @@
...
-			"kind":       "Deployment",
-			"metadata": map[string]interface{}{
-				"name": "deploy1",
-			},
-			"spec": map[string]interface{}{
-				"template": map[string]interface{}{
-					"metadata": map[string]interface{}{
-						"labels": map[string]interface{}{
-							"old-label":     "old-value",
-							"another-label": "foo",
```
Comment (damienr74) [inline k8sdeps/transformer/patch/transformer_test.go]: It seems that adding a label is no longer tested. Any reason why?

Comment (Liujingfang1) [reply k8sdeps/transformer/patch/transformer_test.go]: Missed it. I can add it back.

Comment (Liujingfang1) [reply k8sdeps/transformer/patch/transformer_test.go]: done

**Pull request merged**